src/Loop.php

```diff
@@ -430,3 +455,10 @@ private function __construct()
         // intentionally left blank
     }
 }
+
+// Reset the $level in order to be able to use ::run() inside a shutdown handler
+$f = function()
+{
+    self::$level = 0;
+};
+register_shutdown_function($f->bindTo(null, Loop::class));
```

Comment:
If this was done in a file included immediately by composer this would be much more likely… but still not guaranteed.

Any particular reason for requiring the level to be 0 to call `Loop::run()`? We don't have this restriction for execute, even when passing the loop driver returned from `Loop::get()`.